c7n/resources/shield.py

```diff
@@ -10,20 +10,40 @@
 from c7n.utils import local_session, type_schema, get_retry
 
 
+class ShieldResourceManager(QueryResourceManager):
+
+    def get_arns(self, resources):
+        """Convert Shield ARNs to the format used elsewhere
```

Comment:
the shield arn is distinct from the resource arn, its not clear why we're converting it here to a different format here.

the goal would be to normalize on the specific filter/action for eni would think.

Reply:
Yeah there may be a better way to handle this. I don't have a Shield Advanced subscription to confirm how it reports EIP protections. But we have [recorded test data](https://github.com/cloud-custodian/cloud-custodian/blob/2ee35fad29d3a976e7700d1e6e1f4babc00200a8/tests/data/placebo/test_zone_shield_enable/shield.ListProtections_3.json#L6) that suggests it returns ARNs of the underlying protected resources.

And from the [Slack thread](https://cloud-custodian.slack.com/archives/CAL4P6YE6/p1675203479130419), @StevenGunn mentioned that the `shield-enabled` filter stopped working for EIPs when we changed the EIP `arn_type` from `eip-allocation` to `elastic-ip`. Everywhere _except_ Shield, `elastic-ip` seems to be the correct resource type. So it seemed like we needed a shim on both the list and set ends, to normalize those ARNs.

Whatever the solution, I'd like to get some flight recordings from folks with Shield EIP protections.

Reply:
I think we want a shield enabled subclass specific to eip that normalizes the eni on the way out of shield, ditto for the action to enable, to normalize on the way into shield.

Reply:
@kapilt @ajkerrigan is this more along the line of what you guys are thinking of? https://github.com/cloud-custodian/cloud-custodian/compare/main...darrendao:cloud-custodian:eip-shield?expand=1

Reply:
If you have test cases that work for that (which is hopefully an eip-specific version of [this set-shield test](https://github.com/cloud-custodian/cloud-custodian/blob/620d17df550df718f2f3dc6c044be13604025511/tests/test_route53.py#L15-L40)) that seems fine (and if we have a test case, that should help if it needs any more tweaks).

Thanks for working on this. I initially thought the multiple inheritance approach looked like overkill but any other alternative I tried led to either reimplementing or changing too much existing logic, or overriding the resource manager as I did in the initial pass. So your way looks better :)

Reply:
@ajkerrigan i've added unit test to my code. Should I go ahead and create a PR and go with that? Or do we want to make the change to your fork?

Reply:
I think you've done all the work here, you can create is as your own PR and I'll close this one